### PR TITLE
Add DOS support

### PR DESCRIPTION
**Pull request by wtdcode**
as title.

### CODE REVIEW

**Comment by wtdcode** [comment]
Waiting for @chfl4gs to update Dockerfile and see why TravisCI gets segment fault on Windows.

File: .travis.yml [hunk 1]
```diff
@@ -115,7 +115,7 @@ matrix:
         - unzip -Pinfected al-khaser.bin.zip
         - cd $TRAVIS_BUILD_DIR/tests
       script:
-        - cmd.exe //C '.\test_pe.bat'
```
Comment (xwings) [inline .travis.yml:118]: We need to change this  back ?

File: qiling/os/dos/dos.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            self.exit_point = self.ql.exit_point
+
+        if  self.ql.entry_point is not None:
+            self.ql.loader.elf_entry = self.ql.entry_point
+        else:
+            self.ql.entry_point = self.ql.loader.start_address
+        if not self.ql.shellcoder:
+            try:
+                self.ql.reg.write("ds", self.ql.loader.cs)
+                self.ql.reg.write("es", self.ql.loader.cs)
```
Comment (xwings) [inline qiling/os/dos/dos.py:143]: all the setup code should not be in run.
most of our setup code in loader

File: tests/test_pe.py [hunk 3]
```diff
@@ -17,6 +17,12 @@
 from unicorn.x86_const import *
 
 class PETest(unittest.TestCase):
+
+    def test_pe_dos_8086_hello(self):
+        ql = Qiling(["../examples/rootfs/8086_dos/HI.COM"], "../examples/rootfs/8086_dos")
```
Comment (xwings) [inline tests/test_pe.py:22]: I suggest we  can make it test_dos.py so we can run the same test in mac and linux

File: setup.py [hunk 4]
```diff
@@ -2,19 +2,28 @@
...
-VERSION = ql_version
+VERSION = gb['__version__']
 
 with open('requirements.txt') as f:
     required = f.read().splitlines()
 
 with open("README.md", "r", encoding="utf-8") as ld:
     long_description = ld.read()
 
+if "linux"  in sys.platform:
```
Comment (xwings) [inline setup.py:21]: can we move all the requirement here ? and not setup.py and requirement.txt

File: qiling/loader/utils.py [hunk 5]
```diff
@@ -111,6 +112,14 @@ def getident():
     return arch, ostype, archendian
 
 def ql_pe_check_archtype(path):
+
+    ftype = magic.from_file(path)
```
Comment (xwings) [inline qiling/loader/utils.py:116]: since we are using libmagic, can we just use libmagic for all file type check ?

File: .travis.yml (hunk 1 above)
Comment (chfl4gs) [reply .travis.yml:118]: Timeout increased for crush debugging. Revert to orig.

Comment (wtdcode) [reply .travis.yml]: See commit c5bd30a.

File: qiling/os/dos/dos.py (hunk 2 above)
Comment (wtdcode) [reply qiling/os/dos/dos.py]: See commit 80f568e.

File: tests/test_pe.py (hunk 3 above)
Comment (wtdcode) [reply tests/test_pe.py]: See commit 2cc0303.

File: setup.py (hunk 4 above)
Comment (wtdcode) [reply setup.py:26]: See commits 4818261, 3c6b61b and 81fca85.

File: qiling/loader/utils.py (hunk 5 above)
Comment (wtdcode) [reply qiling/loader/utils.py]: See commit 4795da3.

**Comment by wtdcode** [comment]
@chfl4gs I have updated `Dockerfile` and please rebuild a new image after PR being merged.

In addition, building capstone (4.0.2) from source doesn't work with MacOS. `python3 setup.py install` will always prefer building from sources (since setuptools doesn't support wheels) so I replace it with `pip3 install .` in `.travis.yml` which prefers wheels.

File: .travis.yml [hunk 6]
```diff
@@ -54,7 +52,7 @@ matrix:
 #              sudo pip3 install $HOME/dist/*.whl;
 #          fi
         - cd $TRAVIS_BUILD_DIR 
-        - python3 setup.py install 
+        - pip3 install . # Workaround since building from capstone sources doesn't work on Macos.
```
Comment (xwings) [inline .travis.yml:55]: capstone or keystone

Comment (wtdcode) [reply .travis.yml]: keystone.

Fixed in dbb78be.

**Comment by xwings** [comment]
One more things, please add this feature to ChangeLog

**Pull request merged**